services/website/utils.go

```diff
@@ -140,29 +153,30 @@ func consolidateBuilderEntries(builders []*database.TopBuilderEntry) []*database
 	buildersNumPayloads := uint64(0)
 	for _, entry := range builders {
 		buildersNumPayloads += entry.NumBlocks
-		if strings.Contains(entry.ExtraData, "builder0x69") {
-			topBuilderEntry, isKnown := buildersMap["builder0x69"]
-			if isKnown {
-				topBuilderEntry.NumBlocks += entry.NumBlocks
-				topBuilderEntry.Aliases = append(topBuilderEntry.Aliases, entry.ExtraData)
-			} else {
-				buildersMap["builder0x69"] = &database.TopBuilderEntry{
-					ExtraData: "builder0x69",
-					NumBlocks: entry.NumBlocks,
-					Aliases:   []string{entry.ExtraData},
+		updated := false
+		for k, v := range aliases {
+			// Check if this is one of the known aliases.
+			if v(entry.ExtraData) {
+				updated = true
+				topBuilderEntry, isKnown := buildersMap[k]
+				if isKnown {
+					topBuilderEntry.NumBlocks += entry.NumBlocks
+					topBuilderEntry.Aliases = append(topBuilderEntry.Aliases, entry.ExtraData)
+				} else {
+					buildersMap[k] = &database.TopBuilderEntry{
+						ExtraData: k,
+						NumBlocks: entry.NumBlocks,
+						Aliases:   []string{entry.ExtraData},
+					}
 				}
+				break
 			}
-		} else {
+		}
+		if !updated {
 			buildersMap[entry.ExtraData] = entry
 		}
 	}
 
-	// Prepare top builders by extra stats
```

Comment:
isn't this needed to compute the overall percentage of landed blocks, across all aliases?